### PR TITLE
fix(app): lint templates

### PR DESCRIPTION
**Pull request by pimlie**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR is for the actual linting of the app templates. Currently the errors below still exist, I expect that all `no-unused-vars` (except `interopDefault`) could actually be removed, but as I was not 100% sure I'd like your input on that.

image

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/app/template/utils.js [hunk 1]
```diff
@@ -117,7 +118,6 @@ export async function getRouteData(route) {
 }
 
 export async function setContext(app, context) {
-  const route = (context.to ? context.to : context.route)
```
Comment (pimlie) [inline packages/app/template/utils.js]: Please check this removal with extra care.

File: packages/app/template/client.js [hunk 2]
```diff
@@ -337,21 +337,21 @@ async function render (to, from, next) {
       Component._dataRefresh = false
       // Check if Component need to be refreshed (call asyncData & fetch)
       // Only if its slug has changed or is watch query changes
-      if (this._pathChanged && this._queryChanged || Component._path !== _lastPaths[i]) {
+      if ((this._pathChanged && this._queryChanged) || Component._path !== _lastPaths[i]) {
```
[the remark this answers is not in the document]
Comment (pimlie) [reply packages/app/template/client.js]: Hehe, we actually did the same test more or less :taco: But I tought there could be a small chance this was an unnoticed bug and the wanted behaviour in this line was actually different then the default preferences.

**Review by galvez — APPROVED** [review]

File: packages/common/src/utils.js [hunk 3]
```diff
@@ -389,3 +389,18 @@ const DYNAMIC_ROUTE_REGEX = /^\/(:|\*)/
...
+
+const WHITESPACE_REPLACEMENTS = [
+  [/[ \t\f\r]+\n/g, '\n'], // strip empty indents
+  [/{\n{2,}/g, '{\n'], // strip start padding from blocks
+  [/\n{2,}([ \t\f\r]*})/g, '\n$1'], // strip end padding from blocks
+  [/\n{3,}/g, '\n\n'], // strip multiple blank lines (1 allowed)
+  [/\n{2,}$/g, '\n'] // strip blank lines EOF (0 allowed)
+]
+
+export const stripWhitespace = function stripWhitespace(string) {
```
Comment (pi0) [inline packages/common/src/utils.js]: GREAT!

**Review by pi0 — APPROVED** [review]
You rock @pimlie. However i think there should be better ways of not messing up templates with `<%= isTest ? .... %>` but get along with it for now :)

**Comment by pimlie** [comment]
Thanks for the quick review & merge. 

Most of the isTest's are for when serialize and/or devalue is used, I looked into pretty formatting those JSON structures but tought that would be to cumbersome to do because I couldnt find a formatter which supports a `startIdentation`. [stringify-object](https://github.com/yeoman/stringify-object) comes close, but then still you would need to do a `.split('\n').join('\n' + ' '.repeat(startIndent))` afterwards to have them correctly indented.

**Pull request merged**